pyomo/contrib/doe/doe.py

```diff
@@ -1302,133 +1216,1015 @@ def det_general(m):
+            return model.determinant == det_perm
 
-        if self.Cholesky_option and self.objective_option == ObjectiveLib.det:
-            m.cholesky_cons = pyo.Constraint(
-                m.regression_parameters, m.regression_parameters, rule=cholesky_imp
+        if self.Cholesky_option and self.objective_option == ObjectiveLib.determinant:
+            model.obj_cons.cholesky_cons = pyo.Constraint(
+                model.parameter_names, model.parameter_names, rule=cholesky_imp
             )
-            m.Obj = pyo.Objective(
-                expr=2 * sum(pyo.log10(m.L_ele[j, j]) for j in m.regression_parameters),
+            model.objective = pyo.Objective(
+                expr=2 * sum(pyo.log10(model.L[j, j]) for j in model.parameter_names),
                 sense=pyo.maximize,
             )
 
-        elif self.objective_option == ObjectiveLib.det:
+        elif self.objective_option == ObjectiveLib.determinant:
             # if not cholesky but determinant, calculating det and evaluate the OBJ with det
-            m.det = pyo.Var(initialize=np.linalg.det(fim), bounds=(small_number, None))
-            m.det_rule = pyo.Constraint(rule=det_general)
-            m.Obj = pyo.Objective(expr=pyo.log10(m.det), sense=pyo.maximize)
+            model.determinant = pyo.Var(
+                initialize=np.linalg.det(fim), bounds=(small_number, None)
+            )
+            model.obj_cons.determinant_rule = pyo.Constraint(rule=determinant_general)
+            model.objective = pyo.Objective(
+                expr=pyo.log10(model.determinant + 1e-6), sense=pyo.maximize
+            )
 
         elif self.objective_option == ObjectiveLib.trace:
             # if not determinant or cholesky, calculating the OBJ with trace
-            m.trace = pyo.Var(initialize=np.trace(fim), bounds=(small_number, None))
-            m.trace_rule = pyo.Constraint(rule=trace_calc)
-            m.Obj = pyo.Objective(expr=pyo.log10(m.trace), sense=pyo.maximize)
-            # m.Obj = pyo.Objective(expr=m.trace, sense=pyo.maximize)
+            model.trace = pyo.Var(initialize=np.trace(fim), bounds=(small_number, None))
+            model.obj_cons.trace_rule = pyo.Constraint(rule=trace_calc)
+            model.objective = pyo.Objective(
+                expr=pyo.log10(model.trace), sense=pyo.maximize
+            )
 
         elif self.objective_option == ObjectiveLib.zero:
             # add dummy objective function
-            m.Obj = pyo.Objective(expr=0)
+            model.objective = pyo.Objective(expr=0)
+
+    # Check to see if the model has all the required suffixes
+    def check_model_labels(self, model=None):
+        """
+        Checks if the model contains the necessary suffixes for the
+        DoE model to be constructed automatically.
+
+        Parameters
+        ----------
+        model: model for suffix checking, Default: None, (self.model)
+
+        """
+        if model is None:
+            model = self.model.base_model
```

Comment:
I don't think you need a default here since you always pass in a model every time you use it. Also the docstring didn't match the implementation. The docstring said it would use `self.model` as the default but the implementation was using `self.model.base_model`. In the future, this might be better as a utility function rather than a class method. 
```suggestion
    def check_model_labels(self, model):
        """
        Checks if the model contains the necessary suffixes for the
        DoE model to be constructed automatically.

        Parameters
        ----------
        model: model for suffix checking

        """
```